src/btree/bt_cursor.c

```diff
@@ -960,38 +960,6 @@ __wt_btcur_insert(WT_CURSOR_BTREE *cbt)
     return (ret);
 }
 
-/*
- * __curfile_update_check --
- *     Check whether an update would conflict. This function expects the cursor to already be
- *     positioned. It should be called before deciding whether to skip an update operation based on
- *     existence of a visible update for a key --
- *     even if there is no value visible to the transaction, an update could still conflict.
- */
-static int
-__curfile_update_check(WT_CURSOR_BTREE *cbt)
-{
-    WT_BTREE *btree;
-    WT_PAGE *page;
-    WT_SESSION_IMPL *session;
-
-    btree = cbt->btree;
-    page = cbt->ref->page;
-    session = (WT_SESSION_IMPL *)cbt->iface.session;
-
-    if (cbt->compare != 0)
-        return (0);
-    if (cbt->ins != NULL)
```

Comment:
I hadn't expected this whole function to go away, just that it would only have one call to __wt_txn_update_check

Reply:
I reverted the change. Michael pointed out that it's intentional to have the upd as an argument in __wt_txn_update_check as we may want to check conflict against the new update just appended to the update chain. An example is in __wt_update_serial.